### PR TITLE
Remove callback pattern, #48

### PR DESCRIPTION
**Pull request by geoffdutton**
This removes using the callback provided by Lambda and just returns the value from the app whether it's a promise or not.

Edit: TODO

- [x] Remove callback behavior tests
- [x] Check for breakages for middleware/instances/etc

### CODE REVIEW

File: packages/laconia-core/test/laconia.spec.js [hunk 1]
```diff
@@ -206,30 +201,34 @@ describe("laconia", () => {
 
   describe("callback behaviour", () => {
```
Comment (ceilfors) [inline packages/laconia-core/test/laconia.spec.js]: Most of the tests within this describe block doesn't seem to be related to callback anymore, except the "should not call Lambda callback. Should we pull them out?

Comment (geoffdutton) [reply packages/laconia-core/test/laconia.spec.js]: Agreed.

**Comment by geoffdutton** [comment]
> Looks good! Thanks Geoff!
> 
> I wonder now if this would break any backward compatibility. I hope someone will file a bug, and we can find an alternative way if it breaks compatibility.

Let me look at it one more time tonight, I was just thinking about that last night.

**Comment by ceilfors** [comment]
Hi @geoffdutton, would you still be interested in continuing this PR?

**Comment by geoffdutton** [comment]
@ceilfors Sorry for the delay. I think it should be fine--do you think we're gaining anything with [this test](https://github.com/laconiajs/laconia/blob/5191e0cb12aa42f7bb3821f8298939e1f87336da/packages/laconia-core/test/laconia.spec.js#L227)?

**Comment by ceilfors** [comment]
@geoffdutton Please don't apologise, we're all busy with our own lives! I can't really see what's the change any more after the latest change unfortunately. It seems like there's a merging problem? It looks like this PR is bringing tons of commits from master for some reason.

**Comment by geoffdutton** [comment]
Weird, I think it's because I rebased from master or something. Let me try opening a new PR.
